### PR TITLE
chore/fix: Restructure Dynamo directory [7 / x]

### PR DESCRIPTION
**Pull request by gs-olive**
# Description

- Add `common` directory which stores code common to both the compile and export path, to reduce code duplication and better organize the repository
- Update necessary imports, fix minor argument pass-through issues in `fx_ts_compat`
- Update defaults in `fx_ts_compat` to agree with those in `compile`

## Type of change

Please delete options that are not relevant and/or add your own.

- Chore: restructuring folders, reducing code duplication

# Checklist:

- [ x ] My code follows the style guidelines of this project (You can use the linters)
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ x ] I have made corresponding changes to the documentation
- [ x ] I have added tests to verify my fix or my feature
- [ x ] New and existing unit tests pass locally with my changes
- [ x ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/backend/__init__.py [hunk 1]
```diff
@@ -8,10 +8,10 @@
 from torch_tensorrt import EngineCapability, Device
 from torch_tensorrt.fx.utils import LowerPrecision
 
-from torch_tensorrt.dynamo.backend._settings import CompilationSettings
+from torch_tensorrt.dynamo.common import CompilationSettings
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/backend/__init__.py:11]: do you think its worth having dynamo.common.CompilationSettings vs just dynamo.CompilationSettings?

File: py/torch_tensorrt/dynamo/backend/__init__.py [hunk 2]
```diff
@@ -8,10 +8,10 @@
...
 from torch_tensorrt.dynamo.backend.utils import prepare_inputs, prepare_device
 from torch_tensorrt.dynamo.backend.backends import torch_tensorrt_backend
-from torch_tensorrt.dynamo.backend._defaults import (
+from torch_tensorrt.dynamo.common._defaults import (
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/backend/__init__.py:14]: same here

File: py/torch_tensorrt/dynamo/fx_ts_compat/lower.py [hunk 3]
```diff
@@ -34,21 +45,25 @@ def compile(
...
-    debug=False,
+    explicit_batch_dimension=False,
+    debug=DEBUG,
     refit=False,
     timing_cache_prefix="",
     save_timing_cache=False,
     cuda_graph_batch_size=-1,
     is_aten=False,
-    use_experimental_fx_rt=False,
+    use_experimental_rt=USE_EXPERIMENTAL_RT,
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/fx_ts_compat/lower.py:63]: This would be dependent on the experimental RT PR right?

File: py/torch_tensorrt/dynamo/fx_ts_compat/lower.py [hunk 4]
```diff
@@ -34,21 +45,25 @@ def compile(
...
+    max_aux_streams=MAX_AUX_STREAMS,
+    version_compatible=VERSION_COMPATIBLE,
+    optimization_level=OPTIMIZATION_LEVEL,
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/fx_ts_compat/lower.py:66]: I thought this was already in main?

File: py/torch_tensorrt/dynamo/backend/__init__.py (hunk 1 above)
Comment (gs-olive) [reply py/torch_tensorrt/dynamo/backend/__init__.py:11]: I think it would be better as `dynamo.common.CompilationSettings` until _both_ compile and export use them, since `dynamo.CompilationSettings` might imply it is used across `dynamo`, whereas `common` could just be general purpose utilities.

File: py/torch_tensorrt/dynamo/backend/__init__.py (hunk 2 above)
Comment (gs-olive) [reply py/torch_tensorrt/dynamo/backend/__init__.py:14]: This one, I am more open to moving up a level since the necessary changes have been made to _both_ compile and export. I could add `from torch_tensorrt.dynamo.common._defaults import _defaults` to the `torch_tensorrt.dynamo.__init__.py`

File: py/torch_tensorrt/dynamo/fx_ts_compat/lower.py (hunk 3 above)
Comment (gs-olive) [reply py/torch_tensorrt/dynamo/fx_ts_compat/lower.py:63]: Yes, this is correct.

File: py/torch_tensorrt/dynamo/fx_ts_compat/lower.py (hunk 4 above)
Comment (gs-olive) [reply py/torch_tensorrt/dynamo/fx_ts_compat/lower.py:66]: This was probably accidentally not added in `main` in the PR that added these features to the `TRTInterpreter`

Comment (gs-olive) [reply py/torch_tensorrt/dynamo/fx_ts_compat/lower.py:66]: Split out in #1997

File: py/torch_tensorrt/dynamo/fx_ts_compat/lower.py [hunk 5]
```diff
@@ -122,7 +140,10 @@ def compile(
         save_timing_cache=save_timing_cache,
         cuda_graph_batch_size=cuda_graph_batch_size,
         is_aten=is_aten,
-        use_experimental_rt=use_experimental_fx_rt,
+        use_experimental_rt=use_experimental_rt,
+        max_aux_streams=max_aux_streams,
+        version_compatible=version_compatible,
+        optimization_level=optimization_level,
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/fx_ts_compat/lower.py:146]: Similar fixes in #1997 - to rebase after merge.

**Comment by gs-olive** [comment]
Closed in favor of unified restructuring PR.
